app/server.py

```diff
@@ -49,8 +51,20 @@ def search(search_engine):
                 err = [404, 'No response', qformat]
                 return bad_request(err)
 
-            if((db['queries'].find({query: query}).limit(1)) == False):
-                db['queries'].insert({"query" : query,  "engine" : engine, "qformat" : qformat})
+            flag = 0
+            data = db['queries'].find({"query": query}).limit(1)
```

Comment:
Here `data` contains atmost one document which corresponds to `query` which is not empty.

And below, What I understood, the meaning of for loop is to check whether we have a document or not.

But at the end, you are deleting all documents where `query` is empty, what's the relation of empty query with the flag?